backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/CustomerRetentionRepository.java

```diff
@@ -80,134 +89,111 @@ public I_C_Customer_Retention createNewCustomerRetention(@NonNull final BPartner
 
 		save(customerRetention);
 
-		return customerRetention;
+		return CustomerRetentionId.ofRepoId(customerRetention.getC_Customer_Retention_ID());
 	}
 
-	public void setNewCustomer(@NonNull final BPartnerId bpartnerId)
+	@VisibleForTesting
+	void setNewCustomer(@NonNull final CustomerRetentionId customerRetentionId)
 	{
-		I_C_Customer_Retention customerRetention = retrieveCustomerRetention(bpartnerId);
-
-		if (customerRetention == null)
-		{
-			customerRetention = createNewCustomerRetention(bpartnerId);
-		}
+		final I_C_Customer_Retention customerRetention = getById(customerRetentionId);
 
```

Comment:
please add a comment or better an Check.assumeNotNull here to make it explicit the you made sure elsewhere this won't be null

PS: if you need to rely on something outside this class, such as the model interceptors, i suggest to invoke that getCreate method instead..

Reply:
Check.assumeNotNull done.